### DISCUSSION TITLE
DB 인덱싱에 대한 고찰

### DISCUSSION THREAD

**Discussion by Aleexender** [SCBJ-7/SCBJ-BE#211] [Q&A]
지금 쿼리 튜닝을 진행하고 있는데요!
인덱스에 대한걸 어떻게 걸까 생각중이에요!
사실상 너무 많은 인덱스를 DB에 걸게된다면 insert 할때 성능상의 문제도 그렇고 DB 자체의 메모리도 그렇고 문제되는게 많기에 쉽게 못걸고 있는데요!

#### 제가 인덱스를 걸면서 생각하고있는건 아래와 같습니다
1. 인덱스를 걸어야하는 값은 중복이 많지 않으며 특정 지을수 있는가?(ex모든 제품이 서울로 되어있는데 인덱스를 위치로 잡게되는 경우)
2. 해당 인덱스의 키를 얼마나 많이 사용할 것인가 - 제가 생각하기에 **날짜, 인원과 위치**는 거의 필수적으로 선택할거같아서 이 부분쪽에만 인덱스를 걸 생각입니다!
3. 너무 많은 인덱스를 걸지 않고 최고의 성능을 어떻게하면 끌어올릴수 있을까? - 너무 많은 인덱스를 가지고있다면 속도는 물론 빠르겠지만 공간복잡도에대한 트레이드 오프도 있어서 깊게 생각해봐야하는 부분이라고 생각합니다

제가 생각하는건 위와 같아요! 
혹시라도 어디에 거는게 좋을지 어떻게 거는게 좋을지 아이디어가 있으신분은 적극적인 댓글 부탁드립니당...ㅎ...

https://wiki.yowu.dev/ko/dev/Database/analysis-mysql-query-performance-using-explain-select
제가 보았던 래퍼런스 추가합니당!

**Comment by Kwonminwoo** [comment]
말씀하신 것에 저도 동의합니다! 상황마다 다 다를 것 같아서 확정짓기가 어렵네요
조회하는 테스트 코드에서 시간 체크해보면서 뭐가 더 좋은지 보는 것도 좋을 것 같아요

**Reply by Aleexender** [reply]
시간 체크 해봐서 함 봐볼게용 근데 데이터가 좀 적어서 걱정이긴한데 일단 진행시켜~~~
